### ISSUE TITLE
Need long time to generate text, any body can help?

### ISSUE THREAD

**Issue by hanifabd** [abetlen/llama-cpp-python#138]
I used google colab cpu to try it. this is my code. i am trying to use larger context but this take so long to generate the text

```
llm = Llama(model_path="/content/ggml-vicuna-13b-4bit-rev1.bin", n_ctx=2048, n_threads=20, n_batch=2048)

output = llm(
    "Question: Any suggestion for delicious food? Answer:",
    max_tokens=1024,
    temperature=0.9,
    top_p=0.95,
    repeat_penalty=1.2,
    top_k=50,
    echo=True
)

print(output)
```

**Comment by abetlen** [comment]
How many cpus are in the colab notebook? There's an issue in llama.cpp that causes extreme slowdown when n_threads > cpu_count

**Comment by hanifabd** [comment]
with this code i got 2 cpus
 
![image](https://user-images.githubusercontent.com/45164774/235570921-62565881-111d-4678-9987-a865d9d4f6c9.png)

**Comment by abetlen** [comment]
Correct, if the colab notebook supports openblas then you may be able to speed up eval if you install with `LLAMA_OPENBLAS=1 pip install llama-cpp-python` or if CUDA and cuBLAS are install `LLAMA_CUBLAS=1 pip install llama-cpp-python`

**Comment by hanifabd** [comment]
I have try to reduce n_threads to 1 and 2 but still facing the same issue. then i try to install the openblas in my colab and try to reduce the n_batch to 1024 but still same. the generation took so long even more than 10 minutes

**Comment by mzbac** [comment]
Somehow it is very slow comparing with the native llama.cpp. not very sure why.

**Comment by hanifabd** [comment]
Yuppp, i have try for shorter context but quiet slow. context 32 will generate for about 2-5 minutes. let me inform here if i found something usefull. or maybe if someone know something let me know and share here please. thank you

**Comment by gjmulder** [comment]
Any update?

**Comment by gjmulder** [comment]
#232 looks to be the best issue for analysing performance
